File: Tests/src/pages/SettingsPage.java
package pages;

import java.util.List;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class SettingsPage extends BasePage {
	@FindBy(id = "dateFormat")
	private WebElement dateFormat;

	@FindBy(id = "numberFormat")
	private WebElement numberFormat;

	@FindBy(id = "currency")
	private WebElement currency;

	@FindBy(id = "save")
	private WebElement save;

	@FindBy(id = "cancel")
	private WebElement cancel;

	@FindBy(xpath = "//div[text() = 'Logout']")
	private WebElement logout;
	
	public SettingsPage(WebDriver driver) {
		super(driver);
		PageFactory.initElements(this.driver, this);
	}

	public String getDateFormat() { return getSelectValue(dateFormat); }
	public void setDateFormat(String value) { setSelectValue(dateFormat, value); }
	public List<String> getDateFormats(){ return getSelectOptions(dateFormat); }

	public String getNumberFormat() { return getSelectValue(numberFormat); }
	public void setNumberFormat(String value) { setSelectValue(numberFormat, value); }
	public List<String> getNumberFormats(){ return getSelectOptions(numberFormat); }

	public String getCurrency() { return getSelectValue(currency); }
	public void setCurrency(String value) { setSelectValue(currency, value); }
	public List<String> getCurrencies(){ return getSelectOptions(currency); }

	public SettingsPage set(String dateFormat, String numberFormat, String currency) {
		setDateFormat(dateFormat);
		setNumberFormat(numberFormat);
		setCurrency(currency);
		
		return this;
	}
	
	public DepositPage save() {
		save.click();
		return new DepositPage(driver);
	}
	
	public DepositPage cancel() {
		cancel.click();
		return new DepositPage(driver);
	}
	
	public LoginPage logout() {
		logout.click();
		return new LoginPage(driver);
	}
}

File: Tests/pages/SettingsPage.py
from time import sleep

from selenium.webdriver.common.by import By

from pages.BasePage import BasePage


class SettingsPage(BasePage):
    def __init__(self, driver):
        super().__init__(driver)
        sleep(1)

    def get_date_format_options(self): return self.get_select_options("dateFormat")
    def get_number_format_options(self): return self.get_select_options("numberFormat")
    def get_currency_options(self): return self.get_select_options("currency")

    def get_number_format(self): return self.get_selected_option("numberFormat")

    def set_number_format(self, value):
        self.select_in_dropdown("numberFormat", value)
        return self

    def get_date_format(self): return self.get_selected_option("dateFormat")

    def set_date_format(self, value):
        self.select_in_dropdown("dateFormat", value)
        return self

    def get_currency(self): return self.get_selected_option("currency")

    def set_currency(self, value):
        self.select_in_dropdown("currency", value)
        return self

    def save(self):
        from pages.DepositPage import DepositPage
        self.find_element(By.ID, "save").click()
        return DepositPage(self.driver)

    def cancel(self):
        from pages.DepositPage import DepositPage
        self.find_element(By.ID, "cancel").click()
        return DepositPage(self.driver)

    def logout(self):
        self.find_element(By.XPATH, "//div[text() = 'Logout']").click()
